### ISSUE TITLE
Indent errors show more errors than expected

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#4261]
**Describe the bug**
This issue fixed some indent errors but there are other cases:
https://github.com/microsoft/pylance-release/issues/3656

**To Reproduce**
```python
def a():
    print('a')
        print('b')

def b():
    print('c')
```

**Expected behavior**
There shouldn't be an error on `def b()`

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/19672699/204931649-a7c42042-5562-4753-ba64-82714f6b4875.png)

**Comment by heejaechang** [comment]
```python
def a():
    if True:
    pass

    return

def b():
    pass
```

is another example, we didn't recover properly so  `def b(): ...` is considered as a part of `def a(): ...` so `def b..` is marked as unreachable.

here are parse tree dump
```
[15902] 'd:\dd\Jestspectation\tests\test1.py:1:1' => Module (0,0)-(7,8) =>
  [15917] 'd:\dd\Jestspectation\tests\test1.py:1:1' => Function (0,0)-(7,8) => async:(undefined)
    [15916] 'd:\dd\Jestspectation\tests\test1.py:1:5' => Name (0,4)-(0,5) => 'd:\dd\Jestspectation\tests\test1.py:1:5' (Identifier, (0,4)-(0,5)) {"start":4,"length":1,"type":7,"value":"a"} a
    [15903] 'd:\dd\Jestspectation\tests\test1.py:1:8' => Suite (0,7)-(7,8) =>
      [15910] 'd:\dd\Jestspectation\tests\test1.py:2:5' => If (1,4)-(4,10) =>
        [15904] 'd:\dd\Jestspectation\tests\test1.py:2:8' => Constant (1,7)-(1,11) => True
        [15905] 'd:\dd\Jestspectation\tests\test1.py:2:12' => Suite (1,11)-(4,10) =>
          [15906] 'd:\dd\Jestspectation\tests\test1.py:3:5' => StatementList (2,4)-(2,8) =>
            [15907] 'd:\dd\Jestspectation\tests\test1.py:3:5' => Pass (2,4)-(2,8) =>
          [15908] 'd:\dd\Jestspectation\tests\test1.py:5:5' => StatementList (4,4)-(4,10) =>
            [15909] 'd:\dd\Jestspectation\tests\test1.py:5:5' => Return (4,4)-(4,10) =>
      [15915] 'd:\dd\Jestspectation\tests\test1.py:7:1' => Function (6,0)-(7,8) => async:(undefined)
        [15914] 'd:\dd\Jestspectation\tests\test1.py:7:5' => Name (6,4)-(6,5) => 'd:\dd\Jestspectation\tests\test1.py:7:5' (Identifier, (6,4)-(6,5)) {"start":54,"length":1,"type":7,"value":"b"} b
        [15911] 'd:\dd\Jestspectation\tests\test1.py:7:8' => Suite (6,7)-(7,8) =>
          [15912] 'd:\dd\Jestspectation\tests\test1.py:8:5' => StatementList (7,4)-(7,8) =>
            [15913] 'd:\dd\Jestspectation\tests\test1.py:8:5' => Pass (7,4)-(7,8) =>
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.283, which I just published. It will also be included in a future version of pylance.
